### PR TITLE
Strict null checks -- 88.4%

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Raises strict null check coverage to 88.4%

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/components/fields/schemaFields/widgets/UrlMatchPatternWidget.tsx [hunk 1]
```diff
@@ -99,9 +99,9 @@ const UrlMatchPatternWidget: React.VFC<UrlMatchPatternWidgetProps> = (
         }}
       >
         <ArrayWidget
-          schema={{ items: { type: "string" } }}
           addButtonCaption={addButtonCaption}
           {...props}
+          schema={{ items: { type: "string" } }}
```
Comment (grahamlangford) [inline src/components/fields/schemaFields/widgets/UrlMatchPatternWidget.tsx:104]: Per the typing, `props.schema` could exist, so the manual schema needs to come last

File: src/pageEditor/components/SelectorMatchWidget.tsx [hunk 2]
```diff
@@ -43,9 +43,9 @@ const SelectorMatchWidget: React.VFC<SchemaFieldProps> = (props) => {
       }}
     >
       <ArrayWidget
-        schema={selectorMatchItemSchema}
         addButtonCaption="Add Selector"
         {...props}
+        schema={selectorMatchItemSchema}
```
Comment (grahamlangford) [inline src/pageEditor/components/SelectorMatchWidget.tsx:48]: See https://github.com/pixiebrix/pixiebrix-extension/pull/8851/files#r1678354507

File: src/pageEditor/components/UrlPatternWidget.tsx [hunk 3]
```diff
@@ -66,9 +66,9 @@ const UrlPatternWidget: React.VFC<SchemaFieldProps> = (props) => {
       }}
     >
       <ArrayWidget
-        schema={{ items: urlSchemaProject }}
         addButtonCaption="Add URL Pattern"
         {...props}
+        schema={{ items: urlSchemaProject }}
```
Comment (grahamlangford) [inline src/pageEditor/components/UrlPatternWidget.tsx:71]: See https://github.com/pixiebrix/pixiebrix-extension/pull/8851/files#r1678354507

**Review by fungairino — APPROVED** [review]

File: src/pageEditor/store/analysisManager.ts [hunk 4]
```diff
@@ -217,7 +224,7 @@ async function varAnalysisFactory(
     namespace: StateNamespaces.MOD,
     modComponentRef: {
       modComponentId: activeModComponentFormState.uuid,
-      modId: activeModComponentFormState.modMetadata?.id,
+      modId: activeModComponentFormState?.modMetadata?.id,
```
Comment (twschiller) [inline src/pageEditor/store/analysisManager.ts:227]: activeModComponentFormState is being asserted non-nullish above

https://github.com/pixiebrix/pixiebrix-extension/pull/8851/files#diff-267393957065b5a0e21e395df3650d389e3e21fc943f1b420ec4980fb2af1bd1R213

File: src/pageEditor/components/SelectorMatchWidget.tsx (hunk 2 above)
Comment (twschiller) [reply src/pageEditor/components/SelectorMatchWidget.tsx:48]: Should be added as comments in the code? Or will the type checker prevent the error in the future?

File: src/components/formBuilder/formBuilderHelpers.ts [hunk 5]
```diff
@@ -302,9 +302,7 @@ export const produceSchemaOnUiTypeChange = (
  * @param draft The mutable draft of the RJSF schema
  */
 export const normalizeSchema = (draft: Draft<RJSFSchema>) => {
-  if (isEmpty(draft.schema)) {
-    draft.schema = minimalSchemaFactory();
-  }
+  draft.schema ??= minimalSchemaFactory();
```
Comment (twschiller) [inline src/components/formBuilder/formBuilderHelpers.ts:305]: ❤️

File: src/components/formBuilder/edit/FieldEditor.tsx [hunk 6]
```diff
@@ -133,7 +134,7 @@ const TextAreaFields: React.FC<{ uiOptionsPath: string }> = ({
         }}
         isRequired
       />
-      <Collapse in={showSubmitToolbar}>
+      <Collapse in={showSubmitToolbar ?? false}>
```
Comment (twschiller) [inline src/components/formBuilder/edit/FieldEditor.tsx:137]: Reason to prefer here vs. defaulting early in the useField callsite?

**Review by twschiller — APPROVED** [review]

File: src/pageEditor/store/analysisManager.ts (hunk 4 above)
Comment (grahamlangford) [reply src/pageEditor/store/analysisManager.ts:227]: Thanks. Probably ended up asserting for something else after I fixed this spot

File: src/components/formBuilder/edit/FieldEditor.tsx (hunk 6 above)
Comment (grahamlangford) [reply src/components/formBuilder/edit/FieldEditor.tsx:137]: Defaulting early only works with `undefined` values. `showSubmitToolbar` is `boolean | null`

File: src/pageEditor/components/__snapshots__/UrlPatternWidget.test.tsx.snap [hunk 7]
```diff
@@ -72,11 +72,152 @@ exports[`UrlPatternWidget render single pattern 1`] = `
                       </tr>
                     </thead>
                     <tbody>
+                      <tr>
```
Comment (grahamlangford) [inline src/pageEditor/components/__snapshots__/UrlPatternWidget.test.tsx.snap:75]: Test originally was using the wrong schema

File: src/pageEditor/components/SelectorMatchWidget.tsx (hunk 2 above)
Comment (grahamlangford) [reply src/pageEditor/components/SelectorMatchWidget.tsx:48]: Type checker throws the error

**Pull request merged**